### PR TITLE
Let's get rid of ugly statik

### PR DESCRIPTION
**Pull request by fakeboboliu**
## Intro
go1.16 introduced the embed package, it's the more graceful solution for embedding file into binary.
https://golang.org/pkg/embed/

## Pros
 - The code itself is more graceful
 - It makes the binary smaller slightly

image
(Even with upx packed)

## Oh that's sweet, what's the cost?
Use go1.16, well i'm changed ci configurations already

### CODE REVIEW

**Comment by fatedier** [comment]
Thanks for your contribution.

We plan to do this change in the future.

We'd better to compatible with last two golang versions since not all users has upgraded their golang version.

Can we keep this PR open until go1.17 released?

**Comment by fatedier** [comment]
@fakeboboliu  Go1.17 has been released.

You can rebase your code and push again or i will do this work later.

**Comment by fakeboboliu** [comment]
@fatedier done

File: assets/frpc/embed.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package frpc
```
Comment (fatedier) [inline assets/frpc/embed.go:1]: You'd better not move static files into this directory. Keep it in a independent static directory.

File: assets/assets.go [hunk 2]
```diff
@@ -44,34 +37,11 @@ func Load(path string) (err error) {
 		FileSystem = http.Dir(prefixPath)
 		return nil
 	} else {
-		FileSystem, err = fs.New()
+		FileSystem = http.FS(content)
```
Comment (fatedier) [inline assets/assets.go:40]: We can remove error in return arguments since there won't be any error now.

**Comment by fatedier** [comment]
Can you also edit the Makefile:

```
# compile assets into binary file
file:
    rm -rf ./assets/frps/static/*
    rm -rf ./assets/frpc/static/*
    cp -rf ./web/frps/dist/* ./assets/frps/static
    cp -rf ./web/frpc/dist/* ./assets/frpc/static
    rm -rf ./assets/frps/statik
    rm -rf ./assets/frpc/statik
    go generate ./assets/...
```

to

```
# compile assets into binary file
file:
    rm -rf ./assets/frps/static/*
    rm -rf ./assets/frpc/static/*
    cp -rf ./web/frps/dist/* ./assets/frps/static
    cp -rf ./web/frpc/dist/* ./assets/frpc/static
```

**Comment by fakeboboliu** [comment]
You are right, done.

**Review by fatedier — APPROVED** [review]

**Pull request merged**